lib/providers/nutrition.dart

```diff
@@ -302,6 +302,25 @@ class NutritionPlansProvider with ChangeNotifier {
     await database.deleteEverything();
   }
 
+  /// Saves an ingredient to the cache
+  Future<void> cacheIngredient(Ingredient ingredient, {IngredientDatabase? database}) async {
+    database ??= this.database;
```

Comment:
since we call this directly after receiving results from the search, it could happen that the ingredient is already in the db. We should add a check here.